langchain/src/document_loaders/fs/directory.ts

```diff
@@ -59,11 +60,15 @@ export class DirectoryLoader extends BaseDocumentLoader {
           documents.push(...(await loader.load()));
         }
       } else {
-        // I'm aware some things won't be files,
-        // but they will be caught by the "unknown" handling below.
-        const loaderFactory = this.loaders[extname(file.name)];
-        if (loaderFactory) {
-          const loader = loaderFactory(fullPath);
+        let matchedLoader = null;
+        for (const pattern in this.loaders) {
+          if (minimatch(extname(file.name), pattern)) {
+            matchedLoader = this.loaders[pattern as `.${string}`];
```

Comment:
_nit_: unnecessary as `.${string}` typing.


```suggestion
            matchedLoader = this.loaders[pattern];
```